scripts/set-version.rb

```diff
@@ -0,0 +1,30 @@
+#!/usr/bin/env ruby
+
+# set-version.rb
+#
+# SUMMARY
+#
+#   Sets the Vector version to the passed version. This is used to ensure
+#   the proper version is reflected in Vector's resulting binary.
+
+require_relative "setup"
+
+# The version is expected to be set
+VERSION = ENV.fetch("VERSION")
+
+say("Setting Vector version to #{VERSION}")
+
+# Bump the version in the Cargo.toml file
+cargo_content = File.read("#{ROOT_DIR}/Cargo.toml")
+
+new_cargo_content =
+  cargo_content.sub(
+    /name = "vector"\nversion = "([a-z0-9.-]*)"\n/,
+    "name = \"vector\"\nversion = \"#{VERSION}\"\n"
+  )
+
+File.write("#{ROOT_DIR}/Cargo.toml", new_cargo_content)
+
+success("Cargo.toml updated")
+execute!("cargo check")
```

Comment:
not sure if this is a good idea. I don't think we want CI to touch Cargo.lock. Is there a reason we can't have who ever initiates the release just bump and cargo check locally, then checkin the new Cargo.toml and Cargo.lock?

Reply:
I'd also like to represent the correct version when building nightlies. That will be helpful for debugging customer issues.

Reply:
Is this because on master it is `0.4.0-dev`? If this is the case can we just drop the `-dev` and let nightlies build off of the current version on master?

Reply:
And is there a reason `cargo check` would change anything in the lock file? This is mostly why I was asking how we can update that in a way that reflects only the `vector` version change.

Reply:
Yeah cargo check should bump version of the package in the lock file. Check just attempts to compile, if there is a mismatch between what Cargo.toml has like a package version it will update the lock file, otherwise, check doesn't update it. Ideally, before someone makes the new tag they should bump and build vector to ensure both the lock and the toml are up to date. Then that commit should be tagged.